File: apps/minecraft/spigot/src/main/java/crabcraft/net/crabUtilities/netherportals/PortalSettings.java
package crabcraft.net.crabUtilities.netherportals;

import org.bukkit.Material;
import org.bukkit.block.Block;
import org.jspecify.annotations.Nullable;

import java.util.Set;

/**
 * Immutable snapshot of the {@code custom-nether-portals} config, read fresh
 * from {@code config.yml} on each ignite so {@code /crabutilities reload} takes
 * effect without re-registering the listener.
 *
 * @param frameMaterials blocks that count as portal frame (e.g. obsidian)
 * @param minPortalSize  minimum number of interior blocks for a valid portal
 * @param maxPortalWidth maximum portal width (flat axis span)
 * @param maxPortalHeight maximum portal height (vertical span)
 */
record PortalSettings(Set<Material> frameMaterials, int minPortalSize, int maxPortalWidth, int maxPortalHeight) {

    boolean isPortalFrame(final @Nullable Block block) {
        return block != null && this.frameMaterials.contains(block.getType());
    }
}

File: apps/minecraft/spigot/src/main/kotlin/crabcraft/net/crabUtilities/netherportals/PortalSettings.kt
package crabcraft.net.crabUtilities.netherportals

import org.bukkit.Material
import org.bukkit.block.Block

/** Immutable snapshot of custom portal settings, invalidated on tweaks reload. */
data class PortalSettings(
    private val frameMaterials: Set<Material>,
    private val minPortalSize: Int,
    private val maxPortalWidth: Int,
    private val maxPortalHeight: Int,
) {
    fun frameMaterials() = frameMaterials

    fun minPortalSize() = minPortalSize

    fun maxPortalWidth() = maxPortalWidth

    fun maxPortalHeight() = maxPortalHeight

    fun isPortalFrame(block: Block?) = block != null && frameMaterials.contains(block.type)
}
